### PR TITLE
Add experimental gizmo/server/kit package backed by go-kit & gRPC

### PR DESCRIPTION
**Pull request by jprobinson**
# This is adding a new, experimental package to Gizmo!

* The rationale behind this package:
    * A more opionated server with fewer choices.
    * go-kit is used for serving HTTP/JSON & gRPC is used for serving HTTP2/RPC
    * Monitoring and metrics are handled by a sidecar (ie. Cloud Endpoints)
    * Logs always go to stdout/stderr
    * Using Go's 1.8 graceful HTTP shutdown
    * Services using this package are generally meant for deploying to GCP with GKE and Cloud Endpoints.

To help us figure out what else this server needs (tracing, at least), I'd like to bring this into master and start using it. To deter first-time gizmo users from jumping straight into this server, I'm making sure we label it as 'experimental' for now.

### CODE REVIEW

File: server/kit/router.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+// RouterOption sets optional Router overrides.
+type RouterOption func(Router) Router
+
+// RouterSelect allows users to override the
+// default use of the Gorilla Router.
+// TODO add type and constants for names + docs
+func RouterSelect(name string) RouterOption {
+	return func(_ Router) Router {
```
Comment (fsouza) [inline server/kit/router.go]: Tiny one: should we use just `func(Rounter) Router` instead of `func(_ Router) Router`?

File: server/kit/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# This is an experimental package in Gizmo!
+
+* The rationale behind this package:
+    * A more opionated server with fewer choices.
```
Comment (oliver-nyt) [inline server/kit/README.md]: opionated --> opinionated

File: server/kit/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    * go-kit is used for serving HTTP/JSON & gRPC is used for serving HTTP2/RPC
+    * Monitoring and metrics are handled by a sidecar (ie. Cloud Endpoints)
+    * Logs always go to stdout/stderr
+    * Using Go's 1.8 graceful HTTP shutdown
+    * Services using this package are meant for dpeloy to GCP with GKE and Cloud Endpoints.
```
Comment (oliver-nyt) [inline server/kit/README.md]: dpeloy --> deploy

File: server/kit/kitserver.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	return nil
+}
+
+func (s *server) Start() error {
+	// TODO(jprobinson) add basic default health check
+
+	// HTTP SERVER!
+	addr := fmt.Sprintf(":%d", s.cfg.HTTPPort)
```
Comment (oliver-nyt) [inline server/kit/kitserver.go]: can't configure the full address where this listens?

File: server/kit/router.go (hunk 1 above)
Comment (jprobinson) [inline server/kit/router.go]: 👍

File: server/kit/kitserver.go (hunk 4 above)
Comment (jprobinson) [inline server/kit/kitserver.go]: This is meant to be in a container, DNS should be terminated by the time it gets to this point.

Comment (oliver-nyt) [inline server/kit/kitserver.go]: 🛂

Comment (oliver-nyt) [inline server/kit/kitserver.go]: Ok.

**Pull request merged**